smartsim/_core/_cli/build.py

```diff
@@ -201,23 +196,17 @@ def build_redis_ai(
     ):
 
         # make sure user isn't trying to do something silly on MacOS
-        if self.build_env.PLATFORM == "darwin":
-            if device == "gpu":
-                logger.error("SmartSim does not support GPU on MacOS")
-                sys.exit(1)
-            if onnx and self.versions.REDISAI < "1.2.6":
-                logger.error("RedisAI < 1.2.6 does not support ONNX on MacOS")
-                sys.exit(1)
-            if self.versions.REDISAI == "1.2.4" or self.versions.REDISAI == "1.2.5":
-                logger.error("RedisAI support for MacOS is broken in 1.2.4 and 1.2.5")
-                sys.exit(1)
+        if self.build_env.PLATFORM == "darwin" and device == "gpu":
+            raise BuildError("SmartSim does not support GPU on MacOS")
```

Comment:
We still support RAI 1.2.5, so some of these checks are still needed

Reply:
Also something I should have called out in the PR description!!

We were making two very similar sets of checks in two different location: both here and in in `Versioner`/`RedisAIVersion`. I decided to try and move as many of the RAI compatibility checks into `RedisAIVersioner` to try and eliminate the duplication.

Now when `RedisAIVersion` is instantiated, it will use its knowledge of the platform it is on and the version of python it is running with to try and determine if the RedisAI/ML backends can be installed. If for whatever reason a compatibility inconstancy is found it will raise a `SetupError` immediately. This means that downstream we should be able to use the instanced `RedisAIVersion` without the need to make additional compatibility checks. 

For instance this will now error out on OSX or with Python==3.10:
```sh
$ export SMARTSIM_REDISAI=1.2.5
$ smart build  # same erorr raised at ``pip install`` time as well
# Shortened for brevity
smartsim._core._install.buildenv.SetupError: RedisAI version must be greater than or equal to 1.2.7
```

But the same scenario will work on linux with Python<=3.9
```sh
$ export SMARTSIM_REDISAI=1.2.5
$ smart build
# Shortened for brevity
$ echo $?
0
```

Do let me know if you think this is acceptable/sufficient change! I could definitely be talked into potentially rolling this back if the changes are deemed "too severe"

Reply:
This sounds reasonable to me, but let's bring it up in standup tomorrow and make sure the whole team's onboard